### ISSUE TITLE
BPF lb maps before restart are not the same as after restart

### ISSUE THREAD

**Issue by joestringer** [cilium/cilium#2579]
Testsuite failure seen while debugging bash CI issues:

https://jenkins.cilium.io/job/Cilium-Bash-Tests/116/execution/node/32/log/?consoleFull

```
22:00:26 ==> cilium-master-build-Cilium-Bash-Tests-116: ----- 06-lb.sh: main/test_svc_restore_functionality: checking services and BPF maps after restarting Cilium
22:00:26 ==> cilium-master-build-Cilium-Bash-Tests-116: ----- 06-lb.sh: main/test_svc_restore_functionality: BPF LB maps before restart: 2.2.2.2:0         10.11.251.95:0 (223)           
22:00:26 ==> cilium-master-build-Cilium-Bash-Tests-116: 3.3.3.3:0         10.0.2.15:0 (225)              
22:00:26 ==> cilium-master-build-Cilium-Bash-Tests-116: [f00d::1:1]:0     [f00d::a0f:0:0:f8a]:0 (222)    
22:00:26 ==> cilium-master-build-Cilium-Bash-Tests-116: ----- 06-lb.sh: main/test_svc_restore_functionality: BPF LB maps after restart: 3.3.3.3:0         10.0.2.15:0 (225)              
22:00:26 ==> cilium-master-build-Cilium-Bash-Tests-116: [f00d::1:1]:0     [f00d::a0f:0:0:f8a]:0 (222)    
22:00:26 ==> cilium-master-build-Cilium-Bash-Tests-116: [f00d::1:2]:0     [fd02:1:1:1:1:1:1:1]:0 (224)   
22:00:26 ==> cilium-master-build-Cilium-Bash-Tests-116: ------------------------------------------------------------------------
22:00:26 ==> cilium-master-build-Cilium-Bash-Tests-116:                             Test Failed
22:00:26 ==> cilium-master-build-Cilium-Bash-Tests-116: Error: BPF lb maps before restart are not the same as after restart
22:00:26 ==> cilium-master-build-Cilium-Bash-Tests-116: 
22:00:26 ==> cilium-master-build-Cilium-Bash-Tests-116: ------------------------------------------------------------------------
```

**Comment by ianvernon** [comment]
Is it possible that this commit caused it? 1341c1ba627cbabc4bc2b104ede42746504c6120
There have been no changes to the service / loadbalancer package for some time, so I'm wondering if CLI output formatting changing may be the culprit.

**Comment by joestringer** [comment]
> Is it possible that this commit caused it? 1341c1b
There have been no changes to the service / loadbalancer package for some time, so I'm wondering if CLI output formatting changing may be the culprit.

I'm suspecting this may be the case. I can't tell whether there was a successful CI run for the corresponding PR (https://github.com/cilium/cilium/pull/2547). I don't even understand why it was merged if it has "[WIP]" in the title.

Currently I am trying to reproduce the issue locally.

**Comment by tgraf** [comment]
> I'm suspecting this may be the case. I can't tell whether there was a successful CI run for the corresponding PR (#2547). I don't even understand why it was merged if it has "[WIP]" in the title.

The CI runs were all green and it had two approved reviews from codeowners when I merged it.

**Comment by joestringer** [comment]
Weirdly, it seems like PR #2577 is consistently failing due to this issue, yet when combined with the revert of commit 1341c1b in PR #2582 it passed (at least once, anyway).

**Comment by joestringer** [comment]
I am suspecting that the `tabwriter` makes the ordering of the output in `cilium bpf lb list` less reliable, and that the tests are unintentionally stripping one of the entries from the comparison. Commit https://github.com/cilium/cilium/pull/2577/commits/d45fad40c647d6d9310019410ed80634c9dcddb0 will test my hypothesis. (+ @ianvernon )

**Comment by joestringer** [comment]
Fixed by #2577 .
